### ISSUE TITLE
v12.0.0: Database upgrade sometime fails

### ISSUE THREAD

**Issue by jrouzierinverse** [inverse-inc/packetfence#7283]
Sometimes the database upgrade fails.

```
--------------
ALTER TABLE locationlog
   DROP CONSTRAINT IF EXISTS `locationlog_tenant_id`,
   DROP PRIMARY KEY,
   ADD PRIMARY KEY (`mac`),
   DROP IF EXISTS tenant_id
--------------

ERROR 1025 (HY000) at line 89: Error on rename of './pf/#sql-alter-15754-20' to './pf/locationlog' (errno: 150 "Foreign key constraint is incorrectly formed")
```

**Comment by nqb** [comment]
Workaround we will provide will need to be handle automatically by `do-upgrade.sh`.

**Comment by julsemaan** [comment]
https://www.youtube.com/shorts/uSMUnoHdei8

**Comment by nqb** [comment]
Fixed by #7290

**Comment by nqb** [comment]
There is still an issue with the fire stick, sorry for the inconvenience:

```
ALTER TABLE locationlog
   DROP PRIMARY KEY,
   ADD PRIMARY KEY (`mac`),
   DROP IF EXISTS tenant_id
--------------

ERROR 1553 (HY000) at line 95: Cannot drop index 'PRIMARY': needed in a foreign key constraint
```
@jrouzierinverse, could you look into this ? It has been backported to maintenance so it could have impact on users.
